common/dev/altera.c

```diff
@@ -7,6 +7,18 @@
 #include "hal_i2c.h"
 
 #define MAX_RETRY 3
+#define CHECK_ALTERA_STATUS_DELAY_US 100
+
+// Change bit 1010 1010b(0xaa) and 0101 0101b (0x55)
+// e.g. 0123 4567 -> 1032 5476
+// Change bit 1100 1100b(0xcc) and 0011 0011b(0x33)
+// e.g. 1032 5476 -> 3210 7654
+// Change bit 1111 0000b(0xf0) and 0000 1111b(0x0f)
+// e.g. 3210 7654 -> 7654 3210
+#define SWAP_LSB_TO_MSB(x)                                                                         \
```

Comment:
There is a swap32() function in the /common/dev/pex89000.c file.  Can we move this outside that file and use it for all swaps?

```
static void swap32(uint32_t *d)
{
   if (!d)
     return;

  uint32_t t = ((*d >> 24) & 0xFF) | ((*d >> 8) & 0xFF00) | ((*d << 8) & 0xFF0000) |
  	     ((*d << 24) & 0xFF000000);

  *d = t;
}
```

Reply:
Hi @GoldenBug ,
Thanks for your suggestions.
We check the function and think they are doing different things.

SWAP_LSB_TO_MSB() changes 2 bytes and the minimum unit of change is a bit.
For example, origin data 0xA2(1010 0010) after change is 0x45(0100 0101).
However, swap32() changes 8 bytes, and the minimum unit of change is a word.
For example, origin data 0x100000A2 after change is 0xA2000010.